.github/workflows/haskell.yml

```diff
@@ -22,30 +22,32 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["8.10.7", "9.2.8", "9.4.8", "9.6.5", "9.8.2", "9.10.1"]
-        cabal: ["3.10.3.0"]
-        os: [ubuntu-latest] # ubuntu-latest = ubuntu-22.04
-        liburing: ["liburing-2.6"]
+        ghc: ["8.10.7", "9.2.8", "9.4.8", "9.6.6", "9.8.4", "9.10.1"]
+        cabal: ["3.12.1.0"]
+        os: [ubuntu-latest] # ubuntu-latest = ubuntu-22.04 (soon to be 24.04)
+        liburing: ["system"]  # system liburing is 2.1 on 22.04, 2.5 on 24.04
         include:
-        - ghc: "9.6.5"
-          cabal: "3.10.3.0"
```

Comment:
Omitting these from `include`s means that `haskell-actions/setup@v2` installs the newest cabal version by default. I think it's better to fix the cabal version so that the build doesn't break under our feet when a new version is installed. When we fix all build options we'll have fewer surprises